src/libs/ReportUtils.ts

```diff
@@ -673,128 +594,90 @@ function hasOnlyDistanceRequestTransactions(iouReportID) {
- * @returns {Boolean}
  */
-function isThreadFirstChat(reportAction, reportID) {
-    return !_.isUndefined(reportAction.childReportID) && reportAction.childReportID.toString() === reportID;
+function isThreadFirstChat(reportAction: OnyxEntry<ReportAction>, reportID: string): boolean {
+    return reportAction?.childReportID?.toString() === reportID;
 }
 
 /**
  * Checks if a report is a child report.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isChildReport(report) {
+function isChildReport(report: OnyxEntry<Report>): boolean {
     return isThread(report) || isTaskReport(report);
 }
 
 /**
  * An Expense Request is a thread where the parent report is an Expense Report and
  * the parentReportAction is a transaction.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isExpenseRequest(report) {
-    if (isThread(report)) {
+function isExpenseRequest(report?: OnyxEntry<Report>): boolean {
+    if (report && isThread(report)) {
         const parentReportAction = ReportActionsUtils.getParentReportAction(report);
-        const parentReport = lodashGet(allReports, [`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`]);
-        return isExpenseReport(parentReport) && ReportActionsUtils.isTransactionThread(parentReportAction);
+        const parentReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report?.parentReportID}`];
+        return isExpenseReport(parentReport) && isTypeReportAction(parentReportAction) && ReportActionsUtils.isTransactionThread(parentReportAction);
     }
     return false;
 }
 
 /**
  * An IOU Request is a thread where the parent report is an IOU Report and
  * the parentReportAction is a transaction.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isIOURequest(report) {
-    if (isThread(report)) {
+function isIOURequest(report?: OnyxEntry<Report>): boolean {
+    if (report && isThread(report)) {
         const parentReportAction = ReportActionsUtils.getParentReportAction(report);
-        const parentReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`];
-        return isIOUReport(parentReport) && ReportActionsUtils.isTransactionThread(parentReportAction);
+        const parentReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report?.parentReportID}`];
+        return isIOUReport(parentReport) && isTypeReportAction(parentReportAction) && ReportActionsUtils.isTransactionThread(parentReportAction);
     }
     return false;
 }
 
 /**
  * Checks if a report is an IOU or expense request.
- *
- * @param {Object|String} reportOrID
- * @returns {Boolean}
  */
-function isMoneyRequest(reportOrID) {
-    const report = _.isObject(reportOrID) ? reportOrID : allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportOrID}`];
+function isMoneyRequest(reportOrID: OnyxEntry<Report> | string): boolean {
+    const report = typeof reportOrID === 'object' ? reportOrID : allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportOrID}`];
```

Comment:
@kubabutkiewicz Maybe checking for the `string` will be a little bit safer.
```suggestion
    const report = typeof reportOrID === 'string' ? allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportOrID}`]  :  reportOrID;
```